lib/vector-core/src/event/lua/util.rs

```diff
@@ -51,7 +51,7 @@ pub fn table_to_timestamp(t: LuaTable<'_>) -> LuaResult<DateTime<Utc>> {
     let sec = t.raw_get("sec")?;
     let nano = t.raw_get::<_, Option<u32>>("nanosec")?.unwrap_or(0);
     Ok(Utc
-        .ymd(year, month, day)
-        .and_hms_nano_opt(hour, min, sec, nano)
-        .expect("invalid timestamp"))
+        .with_ymd_and_hms(year, month, day, hour, min, sec)
+        .unwrap()
+        + Duration::nanoseconds(nano as i64))
```

Comment:
Would it make sense to use `with_nanoseconds` here instead of addition? I was playing with something like:
```suggestion
        .with_ymd_and_hms(year, month, day, hour, min, sec)
        .and_then(|t| t.with_nanoseconds(nano))
        .unwrap()
```
Also, please avoid the use of a bare `.unwrap()`, especially where the original code had a `.expect` as this removes the context of the error message that would be generated.

Reply:
Unfortunately the return type of  `with_ymd_amd_hms`, `LocalResult`, is not a well-known rust result. It has no combinators like `and_then` or `expect`.  https://docs.rs/chrono/latest/chrono/offset/enum.LocalResult.html

Reply:
Oh, that's right. I played around with that in my own pull request. `.unwrap()` appears to be equivalent to `.single().unwrap()`, so you can throw in a `.single()` to let you use `expect` or the other combinators.

Reply:
Resolved!